jax/_src/numpy/lax_numpy.py

```diff
@@ -3544,8 +3556,20 @@ def identity(n: DimSize, dtype: DTypeLike | None = None) -> Array:
    To avoid precision errors, consider using an expression like
    ``(jnp.arange(-600, 600) * .01).astype(jnp.bfloat16)`` to generate a sequence in a higher precision
    and then convert it to the desired lower precision.
-""")
+""", extra_params="""
+device : :py:class:`Device`, :py:class:`Sharding`, optional
+  The (optional) :py:class:`Device`, :py:class:`Sharding`,
+  representing the device(s) to which created array should be
+  transferred. If given, then the result is committed to the device(s).
+"""
+)
 def arange(start: DimSize, stop: DimSize | None = None,
+           step: DimSize | None = None, dtype: DTypeLike | None = None,
+           device: xc.Device | Sharding | None = None) -> Array:
+  return jax.device_put(_arange(start, stop=stop, step=step, dtype=dtype), device=device)
```

Comment:
Add a todo here to optimize putting the array directly on the device specified instead of putting it on device 0 and then on the specific device.

Here and above.